src/allmydata/test/test_node.py

```diff
@@ -17,6 +22,14 @@
 import allmydata.test.common_util as testutil
 
 
+# avoid "too many open files" with macOS (only)
+# * this does not seem to be an issue with Linux
+# * appveyor complains about missing 'resource' module on Windows
+if 'Darwin' == system():
+    import resource
+    resource.setrlimit(resource.RLIMIT_NOFILE, (2000, -1))
```

Comment:
It seems like this is probably only necessary because the change to `mkstemp()` results in a leaked file descriptor per call.  If you add a call to close the file descriptor then they won't leak and the process won't run out.

Reply:
I tried it with no success. Currently I can't solve this and would withdraw this PR if there is no silver lining within the next week.